scripts/helpers

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+
+set -Eeuo pipefail
+
+# Validates dev_tags for DockerHub only pushes to "hashicorppreview" org
+dev_tags_validation() {
+  if [ -n "${1}" ]; then
+    for dt in ${1}; do
+      # strip docker.io if exists
+      dt_stripped=${dt//docker.io\//}
+
+      # if there are two parts we presume it's the public docker registry, ex: hashicorppreview/consul. Anything else is a registry we don't check currently.
+      parts=$(echo "$dt_stripped" | awk -F'/' '{print NF}')
+      if [ "$parts" -eq 2 ] && [[ "$dt_stripped" != hashicorppreview/* ]]; then
+        echo "Only valid DockerHub organization is 'hashicorppreview' (Got: $dt)"
+        exit 1
```

Comment:
Ideally this should `return 1` rather than exiting since it's a function. This will give the caller the option to handle the error differently if it needs to. (At the moment, `return 1` would cause the script to exit anyway since we have `set -e` enabled in the caller.)